internal/pkg/deploy/pipeline.go

```diff
@@ -78,82 +75,117 @@ type Source struct {
 	Properties map[string]interface{}
 }
 
+// CodeCommitSource defines the (CC) source of the artifacts to be built and deployed.
+type CodeCommitSource struct {
+	// The name of the source code provider. For example, "GitHub"
+	ProviderName string
+
+	// Contains provider-specific configurations, such as:
+	// "region": "us-west-1"
+	Properties map[string]interface{}
+}
+
 // GitHubPersonalAccessTokenSecretID returns the ID of the secret in the
 // Secrets manager, which stores the GitHub Personal Access token if the
 // provider is "GitHub". Otherwise, it returns the detected provider.
-func (s *Source) GitHubPersonalAccessTokenSecretID() (string, error) {
-	// TODO type check if properties are GitHubProperties?
+func (s *GitHubSource) GitHubPersonalAccessTokenSecretID() (string, error) {
 	secretID, exists := s.Properties[manifest.GithubSecretIdKeyName]
 	if !exists {
 		return "", errors.New("the GitHub token secretID is not configured")
 	}
-
 	id, ok := secretID.(string)
 	if !ok {
 		return "", fmt.Errorf("unable to locate the GitHub token secretID from %v", secretID)
 	}
+	return id, nil
+}
 
-	if s.ProviderName != manifest.GithubProviderName {
-		return fmt.Sprintf("Non-GitHub provider detected: %s", s.ProviderName), nil
+// parseOwnerAndRepo parses the owner and repo name from the GH repo URL, which was formatted and assigned in cli/pipeline_init.go.
+func (s *GitHubSource) parseOwnerAndRepo() (string, string, error) {
+	// NOTE: this is duplicated from validate.go
+	// Ex: https://github.com/koke/grit
+	ghRepoExp := regexp.MustCompile(`(https:\/\/github\.com\/|)(?P<owner>.+)\/(?P<repo>.+)`)
```

Comment:
it might still be best to keep these regexes as package-level variables so that they are initialized once. Otherwise both for `Repository()` and `Owner()` we're recreating them

Reply:
Can I define them at the package-level but locate them just above the method?

Reply:
Sure, although I think the convention is to put at the top of the file but that's fine w me!

Reply:
I'll stick with convention--I had only moved it down bc PH asked it to be closer to where it's used, but since it's package-level, I'll put it back up top.